lti_consumer/api.py

```diff
@@ -67,12 +82,25 @@ def _get_lti_config(config_id=None, block=None):
             lti_config = _get_or_create_local_lti_config(
                 config.get("version"),
                 block.location,
-                block.external_config
+                external_id=block.external_config,
+                config_type="external"
+            )
+        elif block.config_type == "database":
+            lti_config = _get_or_create_local_lti_config(
+                # TODO: I'm not satisfied with this. I want the version to come from the LTIConfiguration model, not
```

Comment:
Any thoughts?

Reply:
this is a version of my question above so you're already thinking of it

_get_or_create doesn't do that much work aside from choosing store type and you already switch on db/block/external here. Would it make sense to have _get_or_create_from_db and _get_or_create_external and _get_or_create_on_xblock which take different parameters? Each of those would be tiny.

Reply:
Definitely a fan of this idea. I think passing around the value of config_type tends to get a bit messy because again it's nicer to have things come from the model and not a string off the xblock.